java/core/src/main/java/com/microsoft/bond/SetBondType.java

```diff
@@ -148,6 +154,21 @@ protected final void serializeValue(SerializationContext context, Set<TElement>
     }
 
     @Override
+    protected final Set<TElement> deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        Set<TElement> value = newDefaultValue(count);
```

Comment:
So, ceiling(count / loadFactor) could be a good initial capacity which can prevent rehashes while the set is being initialized.

---
In reply to: [130754748](https://github.com/Microsoft/bond/pull/542#discussion_r130754748) [](ancestors = 130754748)